source/parser/identifiable_parser/identifiable_parser.hpp

```diff
@@ -10,13 +10,20 @@
 namespace sim {
 
 template <typename T>
-class IdentifieableParser {
+class IdentifieableParser  {
+    static_assert(std::is_base_of_v<Identifiable, T>);
+public:
+    // Parses object and return shared_ptr to it
+    static std::shared_ptr<T> parse_object(const YAML::Node& key_node,
+                                           const YAML::Node& value_node);
+};
+
+template <typename T>
+class IdentifieableRegister {
```

Comment:
Suggestion (might be done only if you accept [this one](https://github.com/Cloud-Storage-Team/algnet/pull/360/files#r2210278182):
```suggestion
class IdentifieableParser {
```